src/components/v5/shared/MembersSelect/MembersSelect.tsx

```diff
@@ -1,41 +1,96 @@
-import React, { FC, useEffect, useState } from 'react';
+import React, { FC, useCallback, useState } from 'react';
+import { Props as ReactSelectProps, SingleValue } from 'react-select';
+
 import { useColonyContext } from '~hooks';
+import { formatText } from '~utils/intl';
+import { DropdownIndicator, SelectBase } from '~v5/common/Fields/Select';
 
-import Select from '~v5/common/Fields/Select';
 import { useGetColonyMembers } from './hooks';
-import { MemberSelectProps } from './types';
+import CustomOption from './partials/CustomOption';
+import { MemberSelectProps, MembersSelectOption } from './types';
+import styles from './MembersSelect.module.css';
 
 const displayName = 'v5.MembersSelect';
 
-const MembersSelect: FC<MemberSelectProps> = ({ user }) => {
+const MembersSelect: FC<MemberSelectProps> = ({
+  onChange: onChangeProp,
+  isSearchable = false,
+  options,
+  defaultValue,
+  ...rest
+}) => {
   const { colony } = useColonyContext();
   const { members, loading } = useGetColonyMembers(colony?.colonyAddress);
-  const [selectedMember, setSelectedMember] = useState<number | undefined>(
-    undefined,
+  const [selectedMember, setSelectedMember] = useState<
+    MembersSelectOption['value'] | undefined
+  >(defaultValue || undefined);
+
+  const selectOptions = members?.reduce<MembersSelectOption[]>(
+    (result, member) => {
+      if (!member) {
+        return result;
+      }
+
+      if (!member.walletAddress || !member.profile) {
+        return result;
+      }
+
+      const { walletAddress, profile } = member;
+
+      return [
+        ...result,
+        {
+          value: walletAddress || '',
+          label: profile?.displayName || walletAddress || '',
+          user: member,
+        },
+      ];
+    },
+    [],
   );
 
-  const handleChange = (selectedOption: number | undefined) => {
-    setSelectedMember(selectedOption);
-  };
+  const onChange = useCallback<
```

Comment:
It should remain `handleChange`, makes it easier to distinguish from the prop too